### PR TITLE
Add QgsAbstractGeometry::asKml

### PR DESCRIPTION
**Pull request by manisandro**

### CODE REVIEW

File: src/core/geometry/qgscurve.cpp [hunk 1]
```diff
@@ -145,6 +145,18 @@ QgsAbstractGeometry *QgsCurve::boundary() const
   return multiPoint;
 }
 
+QString QgsCurve::asKML( int precision ) const
+{
+  QgsLineString *lineString = curveToLine();
```
Comment (m-kuhn) [inline src/core/geometry/qgscurve.cpp]: Better use a unique_ptr

File: src/core/geometry/qgscurvepolygon.cpp [hunk 2]
```diff
@@ -437,6 +437,26 @@ json QgsCurvePolygon::asJsonObject( int precision ) const
   };
 }
 
+QString QgsCurvePolygon::asKML( int precision ) const
+{
+  QString kml;
+  kml.append( "<Polygon>" );
```
Comment (m-kuhn) [inline src/core/geometry/qgscurvepolygon.cpp]: Please use QLatin1String in all the appends

File: src/core/geometry/qgscurvepolygon.cpp [hunk 3]
```diff
@@ -437,6 +437,26 @@ json QgsCurvePolygon::asJsonObject( int precision ) const
...
+  if ( mExteriorRing )
+  {
+    kml.append( "<outerBoundaryIs>" );
+    kml.append( mExteriorRing->asKML( precision ) );
+    kml.append( "</outerBoundaryIs>" );
+  }
+  for ( int i = 0; i < mInteriorRings.size(); ++i )
```
Comment (m-kuhn) [inline src/core/geometry/qgscurvepolygon.cpp]: This should be a range based for loop

File: src/core/geometry/qgsgeometrycollection.cpp [hunk 4]
```diff
@@ -445,6 +445,18 @@ json QgsGeometryCollection::asJsonObject( int precision ) const
   };
 }
 
+QString QgsGeometryCollection::asKML( int precision ) const
+{
+  QString kml;
+  kml.append( "<MultiGeometry>" );
+  for ( int i = 0; i < mGeometries.size(); ++i )
```
Comment (m-kuhn) [inline src/core/geometry/qgsgeometrycollection.cpp]: Range based for loop

File: src/core/geometry/qgspoint.cpp [hunk 5]
```diff
@@ -306,6 +306,11 @@ json QgsPoint::asJsonObject( int precision ) const
   return j;
 }
 
+QString QgsPoint::asKML( int precision ) const
+{
+  return QString( "<Point><coordinates>%1,%2</coordinates></Point>" ).arg( qgsDoubleToString( mX, precision ) ).arg( qgsDoubleToString( mY, precision ) );
```
Comment (m-kuhn) [inline src/core/geometry/qgspoint.cpp]:
```suggestion
  return QStringLiteral( "<Point><coordinates>%1,%2</coordinates></Point>" ).arg( qgsDoubleToString( mX, precision ), qgsDoubleToString( mY, precision ) );
```

**Comment by nirvn** [comment]
Did we not get rid of upper cased acronyms in QGIS 3 to match qt? If so, this function needs to be renamed to asKml().

File: src/core/geometry/qgspoint.cpp [hunk 6]
```diff
@@ -306,6 +306,11 @@ json QgsPoint::asJsonObject( int precision ) const
   return j;
 }
 
+QString QgsPoint::asKML( int precision ) const
+{
+  return QStringLiteral( "<Point><coordinates>%1,%2</coordinates></Point>" ).arg( qgsDoubleToString( mX, precision ) ).arg( qgsDoubleToString( mY, precision ) );
```
Comment (m-kuhn) [inline src/core/geometry/qgspoint.cpp]: Did the proposed single-arg version not work here? This method will be called often and should perform well.

Comment (manisandro) [reply src/core/geometry/qgspoint.cpp]: No was just me who missed it - sorry.

**Review by m-kuhn — APPROVED** [review]
Thanks

File: src/core/geometry/qgsabstractgeometry.h [hunk 7]
```diff
@@ -307,6 +307,12 @@ class CORE_EXPORT QgsAbstractGeometry
      */
     virtual json asJsonObject( int precision = 17 ) SIP_SKIP const;
 
+    /**
+     * Returns a KML representation of the goemetry.
```
[the remark this answers is not in the document]
Comment (manisandro) [reply src/core/geometry/qgsabstractgeometry.h]: Thanks, fixed.

**Pull request merged**